2-cat-gifs edit problem copy (#19)
storacha-network/learnyouw3up: problems/3-delegation/problem.md

Now, lets talk UCAN.

**What are UCANs?**
UCANs (User Controlled Authorization Networks) are like supercharged passwords. They’re used for secure, decentralized authorization. When you create a UCAN, you sign it with your private key, kind of like signing a document with your unique signature.

> **What are UCANs cool?**

Reviewer: ```suggestion
**Why are UCANs cool?**
```

> The best thing about UCANs is that you can securely give permission to others to do things on your behalf without sharing your private key. This is called "delegation."

Reviewer: ```suggestion
The best thing about UCANs is that you can securely give permission to others to do things on your behalf without sharing your private key. This is called "delegation".
```

**Your Mission**

> In this exercise, you're going to delegate an upload/list capability to the workshop. The workshop will create its own private key and pass its public key (DID) to you through stdin (standard input).

Reviewer: ```suggestion
In this exercise, you're going to delegate an `upload/list` capability to the workshop. The workshop will create its own private key and pass its public key (DID) to you via `stdin` (standard input).
```

Then the workshop will be able to **invoke** the capability to list the items _you've_ uploaded to _your_ space. It'll verify it can find that cat gif you uploaded in the previous exercise.

One last thing, make sure the delegation remains valid for **more than one hour**.

**Steps to complete the task**

> **1. Create new file and read DID from stdin**

Reviewer: ```suggestion
**1. Create new file and read a DID from `stdin`**
```

> Create new file, like `ex4.mjs` and then you can read and parse the DID from 'process.stdin':

Reviewer: ```suggestion
Create new file, like `ex4.mjs` and then you can read and parse the DID from `process.stdin`:
```

```js
import fs from 'node:fs'
import * as DID from '@ipld/dag-ucan/did'

const data = fs.readFileSync(process.stdin.fd, 'utf-8')
const principal = DID.parse(data)
```
**2. Create the delegetion**

> Use the client to create a delegation for upload/list to the provided DID:

Reviewer: ```suggestion
Use the client to create a delegation for `upload/list` to the provided DID:
```

```js
import * as Client from '@web3-storage/w3up-client';
```
Create the client and set the expiration time for the delegation:
```js
const client = await Client.create();
const twoHours = 1000 * 60 * 60 * 2; // Two hours in milliseconds
```
In the example above, it is set to two hours.
**3. Generate the delegation**

> Create the delegation for the upload/list capability:

Reviewer: ```suggestion
Create the delegation for the `upload/list` capability:
```
